go/libraries/doltcore/sqle/commit_ancesctors_table.go

```diff
@@ -0,0 +1,134 @@
+
+import (
+	"context"
+
+	"github.com/dolthub/go-mysql-server/sql"
+
+	"github.com/dolthub/dolt/go/libraries/doltcore/doltdb"
+)
+
+var _ sql.Table = (*CommitAncestorsTable)(nil)
+
+// CommitAncestorsTable is a sql.Table implementation that implements a
+// system table which shows (commit, parent_commit) relationships.
+type CommitAncestorsTable struct {
+	dbName string
+	ddb    *doltdb.DoltDB
+}
+
+// NewCommitAncestorsTable creates a CommitAncestorsTable
+func NewCommitAncestorsTable(ctx *sql.Context, dbName string) (sql.Table, error) {
+	ddb, ok := DSessFromSess(ctx.Session).GetDoltDB(dbName)
+	if !ok {
+		return nil, sql.ErrDatabaseNotFound.New(dbName)
+	}
+
+	return &CommitAncestorsTable{dbName: dbName, ddb: ddb}, nil
+}
+
+// Name is a sql.Table interface function which returns the name of the table.
+func (dt *CommitAncestorsTable) Name() string {
+	return doltdb.CommitAncestorsTableName
+}
+
+// String is a sql.Table interface function which returns the name of the table.
+func (dt *CommitAncestorsTable) String() string {
+	return doltdb.CommitAncestorsTableName
+}
+
+// Schema is a sql.Table interface function that gets the sql.Schema of the commit_ancestors system table.
+func (dt *CommitAncestorsTable) Schema() sql.Schema {
+	return []*sql.Column{
+		{Name: "commit_hash", Type: sql.Text, Source: doltdb.CommitAncestorsTableName, PrimaryKey: true},
+		{Name: "parent_hash", Type: sql.Text, Source: doltdb.CommitAncestorsTableName, PrimaryKey: true},
+	}
+}
+
+// Partitions is a sql.Table interface function that returns a partition
+// of the data. Currently the data is unpartitioned.
+func (dt *CommitAncestorsTable) Partitions(*sql.Context) (sql.PartitionIter, error) {
+	return newSinglePartitionIter(), nil
+}
+
+// PartitionRows is a sql.Table interface function that gets a row iterator for a partition.
+func (dt *CommitAncestorsTable) PartitionRows(sqlCtx *sql.Context, _ sql.Partition) (sql.RowIter, error) {
+	return NewCommitAncestorsRowItr(sqlCtx, dt.ddb)
+}
+
+// CommitAncestorsRowItr is a sql.RowItr implementation which iterates over each
+// (commit, parent_commit) pair as if it's a row in the table.
+type CommitAncestorsRowItr struct {
+	ctx   context.Context
+	itr   doltdb.CommitItr
+	ddb   *doltdb.DoltDB
+	cache []sql.Row
+}
+
+// NewCommitAncestorsRowItr creates a CommitAncestorsRowItr from the current environment.
+func NewCommitAncestorsRowItr(sqlCtx *sql.Context, ddb *doltdb.DoltDB) (*CommitAncestorsRowItr, error) {
+	itr, err := doltdb.CommitItrForAllBranches(sqlCtx, ddb)
+	if err != nil {
+		return nil, err
+	}
+
+	return &CommitAncestorsRowItr{
+		ctx: sqlCtx,
+		itr: itr,
+		ddb: ddb,
+	}, nil
+}
+
+// Next retrieves the next row. It will return io.EOF if it's the last row.
+// After retrieving the last row, Close will be automatically closed.
+func (itr *CommitAncestorsRowItr) Next() (sql.Row, error) {
+	if len(itr.cache) == 0 {
+		ch, cm, err := itr.itr.Next(itr.ctx)
+		if err != nil {
+			return nil, err
```

Comment:
Does this handle io.EOF? Add a comment saying so. It looks like this function never return io.EOF otherwise